### DISCUSSION TITLE
fbx loader crashes because it uses int64 meta data entry

### DISCUSSION THREAD

**Discussion by FlorianBorn71** [assimp/assimp#4956] [General]
Hi Team, we have got a customer fbx that crashes because it uses an entry of type __int64 for the user data. This will assert/crash in function FBXConverter::SetupNodeMetadata because there is no counterpart for int64 as it is for uint64:

```
} else if (const TypedProperty<uint64_t> *interpretedUint64 = prop.second->As<TypedProperty<uint64_t>>()) {
            data->Set(index++, prop.first, interpretedUint64->Value());
```
It turns out there is also no entry for AI_INT64 in enum aiMetadataType to begin with, so I wonder what kind of fix you'd suggest.
1) We can do a quick mitigation, where we map to either uint64 or int32 based on the sign of the value
2) Do the full roundtrip with a new enum value in aiMetadataType, if you consider this useful. But I have no good insight about the scale of this change, e.g. all the places where we need to consider a new meta data type.

Thoughts?

**Comment by FlorianBorn71** [comment]
Thanks @kimkulling for implementing!

**Comment by kimkulling** [comment]
You are weolcome. This was just easy to do. And sometimes, when looking at the more than 600 open issues, it is just fine to finish even one task :-).
